src/test/directives/run-async_test.ts

```diff
@@ -0,0 +1,167 @@
+/**
+ * @license
+ * Copyright (c) 2018 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {InitialStateError, runAsync} from '../../directives/run-async.js';
+import {render} from '../../lib/render.js';
+import {html} from '../../lit-html.js';
+import {stripExpressionMarkers} from '../test-utils/strip-markers.js';
+
+const assert = chai.assert;
+
+const hasAbortController = typeof AbortController === 'function';
+const testIfHasAbortController = hasAbortController ? test : test.skip;
+
+suite('runAsync', () => {
```

Comment:
Add tests:
* Multiple dependencies
* Add use of props from render closure in template result functions, and ensure they re-evaluate correctly each render
* Ensure `pending-state` event is not fired in initial state